src/XrdTls/XrdTlsTempCA.cc

```diff
@@ -0,0 +1,315 @@
+        }
+        //m_log.Emsg("CAset", "New CA with hash", fname.c_str(), hash_ptr);
+        m_known_cas.insert(hash_ptr);
+
+        if (XrdCryptosslX509ToFile(ca, outputfp, fname.c_str())) {
+            m_log.Emsg("CAset", "Failed to write out CA", fname.c_str());
+            return false;
+        }
+        ca = chain.Next();
+    }
+    fflush(outputfp);
+
+    return true;
+}
+
+
+class CRLSet {
+public:
+    CRLSet(int output_fd, XrdSysError &err)
+    : m_log(err),
+      m_output_fd(output_fd)
+    {}
+
+    /**
+     * Given an open file descriptor pointing to
+     * a file potentially containing a CRL, process it
+     * for PEM-formatted entries.  If a new, unique CRL
+     * is found, then it is written into the current
+     * tempfile.
+     *
+     * The fname argument is used solely for debugging.
+     *
+     * Returns true on success.
+     */
+    bool processFile(file_smart_ptr &fd, const std::string &fname);
+
+private:
+    XrdSysError &m_log;
+
+        // Grid CA directories tend to keep everything in triplicate;
+        // we keep a unique hash of all known CRLs so we write out each
+        // one only once.
+    std::unordered_set<std::string> m_known_crls;
+    const int m_output_fd;
+};
+
+
+bool
+CRLSet::processFile(file_smart_ptr &fp, const std::string &fname)
+{
+    // Note we purposely leak the outputfp here; we are just borrowing the handle.
+    FILE *outputfp = fdopen(m_output_fd, "w");
+    if (!outputfp) {
+        m_log.Emsg("CAset", "Failed to reopen file for output", fname.c_str());
+        return false;
+    }
+
+    // Assume we can safely ignore a failure to parse; we load every file in
+    // the directory and that will naturally include a number of non-CRL files.
+    for (std::unique_ptr<XrdCryptosslX509Crl> xrd_crl(new XrdCryptosslX509Crl(fp.get(), fname.c_str()));
+         xrd_crl->IsValid();
+         xrd_crl = std::unique_ptr<XrdCryptosslX509Crl>(new XrdCryptosslX509Crl(fp.get(), fname.c_str())))
+    {
+        auto hash_ptr = xrd_crl->IssuerHash(1);
+        if (!hash_ptr) {
+            continue;
+        }
+        auto iter = m_known_crls.find(hash_ptr);
+        if (iter != m_known_crls.end()) {
+            //m_log.Emsg("CRLset", "Skipping known CRL with hash", fname.c_str(), hash_ptr);
+            continue;
+        }
+        //m_log.Emsg("CRLset", "New CRL with hash", fname.c_str(), hash_ptr);
+        m_known_crls.insert(hash_ptr);
+
+        if (!xrd_crl->ToFile(outputfp)) {
+            m_log.Emsg("CRLset", "Failed to write out CRL", fname.c_str());
+            fflush(outputfp);
+            return false;
+        }
+    }
+    fflush(outputfp);
+
+    return true;
+}
+
+}
+
+
+std::unique_ptr<XrdTlsTempCA::TempCAGuard>
+XrdTlsTempCA::TempCAGuard::create(XrdSysError &err) {
+    char ca_fname[] = "/tmp/xrootd_ca_file.XXXXXX.pem";
+    int ca_fd = mkstemps(ca_fname, 4);
+    if (ca_fd < 0) {
+        err.Emsg("TempCA", "Failed to create temp file:", strerror(errno));
```

Comment:
Referencing line 209; both messages are identical eventhough they actually refer to different files. To avoid confusion I suggest the message be:
"Failed to create CA temp file:"